cfgov/unprocessed/js/molecules/Multiselect.js

```diff
@@ -0,0 +1,434 @@
+      _list.children[i].classList.remove( 'filter-match' );
+    }
+
+    _filtered = _optionData;
+  };
+
+  function _filterResults() {
+    _list.classList.add( 'filtered' );
+    var item;
+
+    if ( _filtered.length > 0 ) {
+      _filtered.forEach( function( option, i ) {
+        item = _list.querySelector( 'li[data-option="' + option.value + '"]' );
+
+        item.classList.add( 'filter-match' );
+      } );
+    } else {
+      _noResults();
+    }
+  };
+
+  function _noResults() {
+    _list.classList.add( 'no-results' );
+    _list.classList.remove( 'filtered' );
+  };
+
+  function _bindEvents() {
+    var inputs = _list.querySelectorAll( 'input' );
+
+    $.bind( _search, {
+      input: function() {
+        _evaluate( this.value );
+      },
+      focus: function() {
+        expand();
+      },
+      blur: function() {
+        if ( !_isBlurSkipped ) {
+          collapse();
+        }
+      },
+      mousedown: function() {
+        if ( _fieldset.getAttribute( 'visibility' ) === 'hidden' ) {
+          expand();
+        }
+      },
+      keydown: function( event ) {
+        var key = event.keyCode;
+
+        if ( _fieldset.getAttribute( 'visibility' ) === 'hidden' ) {
+          expand();
+        }
+
+        if ( key === KEY_RETURN ) {
+          event.preventDefault();
+          _highlight( DIR_NEXT );
+        } else if ( key === KEY_ESCAPE ) {
+          _resetSearch();
+          collapse();
+        } else if ( key === KEY_DOWN ) {
+          _highlight( DIR_NEXT );
+        } else if ( key === KEY_TAB
+                    && !event.shiftKey
+                    && _fieldset.getAttribute( 'visibility' ) === 'visible' ) {
+          collapse();
+        }
+      }
+    } );
+
+    $.bind( _list, {
+      mousedown: function() {
+        _isBlurSkipped = true;
+      },
+      keydown: function( event ) {
+        var key = event.keyCode;
+        var checked = event.target.checked;
+
+        if ( key === KEY_RETURN ) {
+          event.preventDefault();
+          event.target.checked = !checked;
+
+          $.fire( _queryOne( event.target ), 'change', {} );
+        } else if ( key === KEY_ESCAPE ) {
+          _search.focus();
+          collapse();
+        } else if ( key === KEY_UP ) {
+          _highlight( DIR_PREV );
+        } else if ( key === KEY_DOWN ) {
+          _highlight( DIR_NEXT );
+        }
+      }
+    } );
+
+    for ( var i = 0, len = inputs.length; i < len; i++ ) {
+      $.bind( inputs[i], {
+        change: function( event ) {
+          _updateSelections( event.target.id );
+          _resetSearch();
+        }
+      } );
+    }
+  };
+
+  function _filterContains( text, input ) {
+    return RegExp( $.regExpEscape( input.trim() ), 'i' ).test( text );
+  };
+
+  function _queryOne( expr, con ) {
+    return typeCheckers.isString( expr ) ?
+      ( con || document ).querySelector( expr ) :
+      expr || null;
+  };
+
+  $.create = function( tag, options ) {
```

Comment:
Sorry, could you update these to stop decorating jquery, or add a TODO? These just need to be updated from, e.g., 

```
$.create = function( tag, options ) {
```

To

```
function _create( tag, options ) {
```

And the call updated from `$.create(…)` to `_create(…)`.

The only reason jquery is being used here is as a vehicle to pass custom methods around, which isn't needed and means they're ending up globally. But I know for the a big PR push it can be nice to push through a working implementation and then clean it up in another PR, so it's fine if you want to do it that way instead.